lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -452,6 +464,37 @@ def __get_k8s_restart_policy(self, job_wrapper):
         """The default Kubernetes restart policy for Jobs"""
         return "Never"
 
+    def __get_volume_mounts_for_job(self, job_wrapper):
+        if self.runner_params.get('k8s_data_volume_claim'):
+            try:
+                param_claim = self.runner_params['k8s_data_volume_claim'].split(":")
+                claim_name = param_claim[0]
+                base_mount = param_claim[1]
+            except Exception as e:
+                log.debug('Failed to parse `k8s_data_volume_claim` parameter in the kubernetes runner configuration')
+                raise e
+            inputs = job_wrapper.get_input_fnames()
```

Comment:
This won't work with extra files paths and metadata files I assume.

This logic shouldn't be happening at this layer.